### ISSUE TITLE
S1172 false positive when function parameter is used inside local function only

### ISSUE THREAD

**Issue by volgunin** [SonarSource/sonar-dotnet#3305]
### Description

When a function parameter is used inside local function only, S1172 is raised.

The code snippet here is not designed to promote any coding patterns and has no much meaning; its only purpose is to reproduce the false positive.

### Repro steps
```C#
        private void Test(int i, int j)
        {
            int[] x = new int[10];

            Swap(x);

            void Swap<T>(T[] array)
            {
                T t = array[i];
                array[i] = array[j];
                array[j] = t;
            }
        }
````
### Expected behavior

No warning.

### Actual behavior

warning S1172: Remove this parameter 'i', whose value is ignored in the method.
warning S1172: Remove this parameter 'j', whose value is ignored in the method.

### Related information

SonarAnalyzer.CSharp.8.6.0.16497
VisualStudio 2019 16.5.4

**Comment by pavel-mikula-sonarsource** [comment]
Hi @volgunin ,

Thank you for reporting this. We already have same scenario reported in #3134 so I'll close this issue.
